app/chapters/[slug]/[lesson]/page.tsx

```diff
@@ -18,5 +21,9 @@ export default function Page({ params }) {
 
   const Lesson = lessons[params.lesson].default
 
-  return <Lesson />
+  return (
+    <IntlProvider>
```

Comment:
The provider probably should just go in the top level layout file, but I couldn't figure out how to get that to work with Next.js app directory.